fancy_collections/formatting.py

```diff
@@ -45,25 +47,50 @@ def key_to_string(self, key: Any, obj: Any) -> str:  # noqa
     def to_string(self) -> str:
         if len(self._obj) == 0:
             return self._stringify_empty_class(self._obj)
+
+        objects = {}
+        widths = {}
         for key, val in self._obj.items():
             key = self.key_to_string(key, val)
-            try:
-                string = self.stringify(val)
-            except Exception:  # noqa
-                string = NotImplemented
-            if string is NotImplemented:
-                string = str(val)
-            self._add(key, string.splitlines())
+            lines = self.stringify(val).splitlines()
+            objects[key] = lines
+            widths[key] = self._get_maxlen(lines + [key])
+
+        # take the potential placeholder into acount
+        display_width = int(shutil.get_terminal_size().columns) - 3 - len(self.column_seperator)
+
+        keys = tuple(widths.keys())
+        front_keys, back_keys = set(), set()
+        line_length = 0
+        for fkey, bkey in zip(keys, keys[::-1]):
+            line_length += widths[fkey] + len(self.column_seperator)
+            if line_length < display_width:
+                front_keys.add(fkey)
+            line_length += widths[bkey] + len(self.column_seperator)
+            if line_length < display_width:
+                back_keys.add(bkey)
+
+        for k, v in objects.items():
+            if k in front_keys:
+                self._add(k, v)
+
+        if not front_keys.intersection(back_keys) and front_keys.union(back_keys) != set(keys):
+            self._add("...", ["..."])
```

Comment:
maybe

```suggestion
        if len(front_keys) + len(back_keys) < len(keys):
            self._add("...", ["..."])
```

Reply:
That is actually not enough, as `back_keys` and `front_keys` are not necessarily disjunctive. In case of few keys or large terminals we might end up with `front_keys == end_keys`, so we need to check the content of both sets.

Reply:
i dont understand your comment. Here we just check the length of the sets.. 

- case 1: if some keys are left and some are right, we must fill the `...` 
- case 2: if left keys and right keys are all keys we dont need the `...`
- case 3: if we have an overlap we have more than total keys and we also don't need the `...`

or do i miss something ?